File: EconRentCarWeb/EconRentCar.DataModel/Renta.cs
using EconRentCar.Common;
using EconRentCar.Core;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EconRentCar.DataModel
{
    public class Renta : IEntityBase
    {
        public Renta()
        {
            Inspecciones = new HashSet<Inspeccion>();
        }
        public Guid Id { get; set; }
        public DateTime FechaRenta { get; set; }
        public DateTime FechaDevolucion { get; set; }
        [Required]
        public string Comentario { get; set; }
        public EstadoRenta EstadoRenta { get; set; }
        public Guid VehiculoId { get; set; }
        public virtual Vehiculo Vehiculo { get; set; }
        public Guid EmpleadoId { get; set; }
        public virtual Empleado Empleado { get; set; }
        public Guid ClienteId { get; set; }
        public virtual Cliente Cliente { get; set; }
        public virtual ICollection<Inspeccion> Inspecciones { get; set; }
    }
}

File: EconRentCarWeb/EconRentCarClient/src/app/shared/models/renta.ts
import { BaseData } from '../dynamic-crud/models';
import { EstadoRenta } from './enums';
import { Vehiculo } from './vehiculo';
import { Empleado } from './empleado';
import { Cliente } from './cliente';
import { Inspeccion } from './inspeccion';

export class Renta implements BaseData {
    public Id = '00000000-0000-0000-0000-000000000000';
    public FechaRenta = new Date();
    public FechaDevolucion = new Date();
    public Comentario = '';
    public EstadoRenta = EstadoRenta.Estacionado;
    public VehiculoId = '00000000-0000-0000-0000-000000000000';
    public Vehiculo: Vehiculo;
    public EmpleadoId = '00000000-0000-0000-0000-000000000000';
    public Empleado: Empleado;
    public ClienteId = '00000000-0000-0000-0000-000000000000';
    public Cliente: Cliente;
    public Inspecciones: Inspeccion[];
}
